packages/commands/src/components/command-menu.js

```diff
@@ -187,45 +183,34 @@ function CommandInput( { isOpen, search, setSearch } ) {
  * @ignore
  */
 export function CommandMenu() {
-	const { registerShortcut } = useDispatch( keyboardShortcutsStore );
 	const [ search, setSearch ] = useState( '' );
-	const isOpen = useSelect(
-		( select ) => select( commandsStore ).isOpen(),
-		[]
-	);
+	const isOpenRef = useRef( false );
+	const isOpen = useSelect( ( select ) => {
+		return ( isOpenRef.current = select( commandsStore ).isOpen() );
+	}, [] );
 	const { open, close } = useDispatch( commandsStore );
 	const [ loaders, setLoaders ] = useState( {} );
 
 	useEffect( () => {
-		registerShortcut( {
-			name: 'core/commands',
-			category: 'global',
-			description: __( 'Open the command palette.' ),
-			keyCombination: {
-				modifier: 'primary',
-				character: 'k',
-			},
-		} );
-	}, [ registerShortcut ] );
+		const handleUserKeyPress = ( e ) => {
+			if ( e.metaKey && e.key === 'k' ) {
+				// Prevent the Save dialog to open
+				e.preventDefault();
 
-	useShortcut(
-		'core/commands',
-		/** @type {import('react').KeyboardEventHandler} */
-		( event ) => {
-			// Bails to avoid obscuring the effect of the preceding handler(s).
-			if ( event.defaultPrevented ) return;
-
-			event.preventDefault();
-			if ( isOpen ) {
-				close();
-			} else {
-				open();
+				if ( isOpenRef.current ) {
+					close();
+				} else {
+					open();
+				}
 			}
-		},
-		{
-			bindGlobal: true,
-		}
-	);
+		};
+
+		document.addEventListener( 'keydown', handleUserKeyPress );
+
+		return () => {
+			document.removeEventListener( 'keydown', handleUserKeyPress );
+		};
+	}, [ close, open ] );
```

Comment:
Can you clarify why you changed this file? I feel it should stay as it was.

Reply:
The useShortcut hooks only works in the editor context (with the provider added) unfortunately. So in order to catch the combination of keys everywhere in WordPress I had to have a global listener instead of the useShortcut.

Reply:
The keyboard-shortcuts package have been updated recently to not require the provider, so it should work in all pages, if it doesn't than we're missing some bug that needs to be fixed.

Reply:
I'll give this another try then. Clearly I would prefer something standardize.

Reply:
Reverted on [0096521](https://github.com/WordPress/gutenberg/pull/54515/commits/00965219c58a130567052d144b9e2d712cd8a398)